lib/Basics/SizeLimitedString.h

```diff
@@ -0,0 +1,181 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <cstring>
+#include <string_view>
+#include <type_traits>
+
+#include "Basics/Endian.h"
+#include "Basics/StringUtils.h"
+#include "Basics/debugging.h"
+
+namespace arangodb {
+
+// small utility class providing a fixed-size string.
+// the maximum length of the string is provided at compile time.
+// data can be added to the string until the fixed capacity is exceeded.
+// from that point on, all further append operations will do nothing
+// (i.e. not append to the buffer or overrun it), nor throw an exception.
+// all operations of this class are noexcept.
+template<size_t N>
+class SizeLimitedString {
+  static_assert(N > 0);
+
+ public:
+  SizeLimitedString(SizeLimitedString const&) = delete;
+  SizeLimitedString& operator=(SizeLimitedString const&) = delete;
+
+  SizeLimitedString() noexcept { clear(); }
+
+  std::string_view view() const noexcept {
+    TRI_ASSERT(_offset <= N);
+    return std::string_view(&_buffer[0], _offset);
+  }
+
+  void clear() noexcept {
+    memset(&_buffer[0], 0, sizeof(_buffer));
+    _offset = 0;
+    _full = false;
+  }
+
+  bool empty() const noexcept { return _offset == 0; }
+
+  size_t capacity() const noexcept { return N; }
+
+  size_t size() const noexcept {
+    TRI_ASSERT(N >= _offset);
+    return _offset;
+  }
+
+  void push_back(char c) noexcept {
+    if (_full) {
+      return;
+    }
+    _buffer[_offset] = c;
+    ++_offset;
+    recalculateState();
+  }
+
+  SizeLimitedString& append(std::string_view data) noexcept {
+    return append(data.data(), data.size());
+  }
+
+  SizeLimitedString& append(char const* data) noexcept {
+    return append(data, strlen(data));
+  }
```

Comment:
This can be removed. `std::string_view` has an implicit conversion constructor from `char*`. In case of string literals this is even performed at compile time.